lib/action_view/component/base.rb

```diff
@@ -117,19 +117,19 @@ def template_file_path
 
           filename = self.instance_method(:initialize).source_location[0]
           filename_without_extension = filename[0..-(File.extname(filename).length + 1)]
-          sibling_files = Dir["#{filename_without_extension}.*"] - [filename]
+          sibling_template_files = Dir["#{filename_without_extension}.????.{#{ActionView::Template.template_handler_extensions.join(",")}}"] - [filename]
```

Comment:
Thoughts on using single quotes within the double quotes to make clear that you aren't stopping the outer double quotes?

Also, TIL `????`
```suggestion
          sibling_template_files = Dir["#{filename_without_extension}.????.{#{ActionView::Template.template_handler_extensions.join(',')}}"] - [filename]
```